server/protocols/jmap-rfc8621-integration-tests/jmap-rfc8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EchoMethodContract.scala

```diff
@@ -40,20 +134,53 @@ trait EchoMethodContract {
   }
 
   @Test
-  def echoMethodShouldReturn200Status(): Unit = {
-    RestAssured
+  def echoMethodShouldResponseOKWithRFC8621VersionAndSupportedMethod(): Unit = {
+    val response: String = RestAssured
       .`given`()
-      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
-      .body(REQUEST_OBJECT)
+        .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+        .body(REQUEST_OBJECT)
       .when()
-      .post()
+        .post()
+        .prettyPeek()
       .then
-      .statusCode(HttpStatus.SC_OK)
-      .contentType(ContentType.JSON)
+        .statusCode(HttpStatus.SC_OK)
+        .contentType(ContentType.JSON)
+      .extract()
+        .body()
+        .asString()
 
+    assertEquals(response, Json.stringify(Json.parse(RESPONSE_OBJECT)))
+  }
 
+  @Test
+  def echoMethodShouldResponseWithRFC8621VersionAndUnsupportedMethod(): Unit = {
+    val response: String = RestAssured
+      .`given`()
+        .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+        .body(REQUEST_OBJECT_WITH_UNSUPPORTED_METHOD)
+      .when()
+        .post()
+        .prettyPeek()
+      .then
+        .statusCode(HttpStatus.SC_OK)
+        .contentType(ContentType.JSON)
+      .extract()
+        .body()
+        .asString()
 
+    assertEquals(response, Json.stringify(Json.parse(RESPONSE_OBJECT_WITH_UNSUPPORTED_METHOD)))
   }
 
+  @Test
+  def echoMethodShouldReturn400StatusWithUnSupportedVersion(): Unit = {
```

Comment:
Is it really necessary to test this? For wrong versions, the Y already tests that behavior in a more general fashion, which should be enough

Reply:
I think it necessary for integration test.

Reply:
Then this is not related to EchoMethod, then it should be tested in another contract/class ;-)

Reply:
So, we keep it on the Y test or here?

Reply:
The Y. The test if the version test is valid is specific to the Y. Or are you gonna test that for all methods? Would make no sense IMO